nefertari_es/fields.py

```diff
@@ -1,30 +1,148 @@
-from datetime import datetime
+import datetime
 from dateutil import parser
-from elasticsearch_dsl import (
-    Date,
-    Field,
-    String,
-    Integer,
-    )
 
+from elasticsearch_dsl import field
+from elasticsearch_dsl.exceptions import ValidationException
 
-DateField = Date
-StringField = String
-TextField = String
-IntField = Integer
+# ChoiceField
+# PickleField
+# TimeField
+# Relationship
+# IdField
+# ForeignKeyField
+# ListField
 
 
-class DateTimeField(Field):
+class CustomMappingMixin(object):
+    """ Mixin that allows to define custom ES field mapping.
+
+    Set mapping to "_custom_mapping" attribute. Defaults to None, in
+    which case default field mapping is used. Custom mapping extends
+    default mapping.
+    """
+    _custom_mapping = None
+
+    def to_dict(self, *args, **kwargs):
+        data = super(CustomMappingMixin, self).to_dict(*args, **kwargs)
+        if self._custom_mapping is not None:
+            data.update(self._custom_mapping)
+        return data
+
+
+class BaseFieldMixin(object):
+    def __init__(self, *args, **kwargs):
+        self._primary_key = kwargs.pop('primary_key', False)
+        super(BaseFieldMixin, self).__init__(*args, **kwargs)
+
+
+class IdField(CustomMappingMixin, field.String):
+    name = 'idfield'
+    _custom_mapping = {'type': 'string'}
+
+    # TODO: Mirror document.meta['_id'] value
```

Comment:
How will we do this? I'm not sure if the field has access to the model. I need to investigate this more.